client/src/app/routes/dataset/analysis/analysis.component.ts

```diff
@@ -327,10 +327,14 @@ export class AnalysisComponent implements OnInit, OnDestroy
     private createTopRowComponents(selectors: string[])
     {
         // Should be 2 components... if not, show the default 2
-        if(selectors.length != 2)
+        if(selectors.length < 2)
         {
             selectors = [ViewStateService.widgetSelectorContextImage, ViewStateService.widgetSelectorSpectrum];
         }
+        else if(selectors.length > 2)
```

Comment:
What happens if selectors > 2? Are we going to allow more widgets up top or is this a bug?

Reply:
When I was adding the new preview widget, I got it into a state for a bit where there were 3 selectors in the top row and this got saved to my view state, which caused it previously to just revert to the default selectors (previous logic was to do that if `length != 2`). I changed it to instead only revert to the default selectors if there were less than 2 selectors stored in the user view state, or if there were more than 3, then just use the first 2. While this probably won't affect anyone, I figured if someone had a malformed view state like I did, it'd be better to recover the valid parts than do a complete reset 🤷‍♂️